src/routes/hydra.ts

```diff
@@ -1,162 +1,209 @@
 import {NextFunction, Request, Response} from 'express'
-import config, {logger} from '../config'
-import {PublicApi, AdminApi as KratosAdminApi, Session } from '@oryd/kratos-client'
-import {AdminApi as HydraAdminApi, AcceptConsentRequest, AcceptLoginRequest, RejectRequest} from '@oryd/hydra-client'
-import url from 'url';
+import config, {
+  logger,
+  SECURITY_MODE_STANDALONE,
+} from '../config'
+import {PublicApi, Session} from '@oryd/kratos-client'
+import {
+  AdminApi as HydraAdminApi,
+  AcceptConsentRequest,
+  AcceptLoginRequest,
+  RejectRequest,
+} from '@oryd/hydra-client'
+
+// This type is used for asserting the "ORY Hydra Consent Context" type.
+// It's simply the ORY Kratos session.
+interface LoginContext extends Session {
+}
 
 const hydraAdminEndpoint = new HydraAdminApi(process.env.HYDRA_ADMIN_URL)
 const kratosPublicEndpoint = new PublicApi(config.kratos.public)
-const kratosAdminEndpoint = new KratosAdminApi(config.kratos.admin)
 
-type UserRequest = Request & { user: any }
+export const hydraLogin = (req: Request, res: Response, next: NextFunction) => {
+  // The kratosRequest is used to identify the login and registration kratosRequest in ORY Kratos and return data like the csrf_token and so on.
+  const kratosRequest = req.query.request
 
-export default (
-  req: Request,
-  res: Response,
-  next: NextFunction
-) => {
-  // The request is used to identify the login and registration request in ORY Kratos and return data like the csrf_token and so on.
-  const request = req.query.request
-  const currentLocation = `${req.protocol}://${req.headers.host}${req.url}`;
-
-  const query = url.parse(req.url, true).query;
-  
-  let challenge = query.login_challenge || req.cookies.login_challenge;
-  if (challenge != null && challenge != "undefined") {
-    logger.debug("Writing the login_challenge cookie from hydra: ", challenge);
+  const hydraChallenge =
+    req.query.login_challenge || req.cookies.login_challenge
+  if (!hydraChallenge) {
+    logger.debug(
+      'Writing the login_challenge cookie from hydra: ',
+      hydraChallenge
+    )
   } else {
-    logger.debug("challenge exists: "+challenge)
-  } 
-  const kratos_session = req.cookies.ory_kratos_session
-
-  if (!request) {
```

Comment:
This were too many nested ifs which makes the code hard to read. I reduced the nesting (see below) and found some issues because of that as well.